lib/DDG/Meta/Data.pm

```diff
@@ -122,6 +125,29 @@ sub get_ia {
     return $m;
 }
 
+sub _satisfy {
+	my ($ia, $by, $lookup) = @_;
+	return $by->($ia, $lookup) if ref $by eq 'CODE';
+	my $pby = $ia->{$by};
+	ref $lookup eq 'CODE' ? $lookup->($pby) : $pby eq $lookup;
+}
+
+# filter_ias({ repo => 'goodies', dev_milestone => 'live'... })
+# Lookups combine as an AND operation.
+sub filter_ias {
+	my ($lookups) = @_;
+	my @results = @ia_container;
+	while (my ($by, $lookup) = each %$lookups) {
```

Comment:
Can you give an example? Currently we reduce the list size on each iteration to only the IAs that have satisfied all previous conditions - where *exactly* would you expect the short-circuit to go?